pkg/azurelustre/nodeserver.go

```diff
@@ -81,6 +82,12 @@ func (d *Driver) NodePublishVolume(
 	if req.GetReadonly() {
 		mountOptions = append(mountOptions, "ro")
 	}
+	for _, userMountFlag := range userMountFlags {
```

Comment:
I think if we do the for-loop before line 82, we don't need to check inside the loop for the read-only option.  We can check once for existence before we try and append on line 82.

Reply:
Sorry, I'm not quite get your point. If we do the for-loop before check req.GetReadonly() and don't check the "ro" option. We still need to check if "ro" exists in the mountOptions when we check req.GetReadonly(). Otherwise, we may add two "ro" to the mountOptions.

Reply:
You still need to check before adding req.ReadOnly.  If you move the for-loop to before that check, the loop no longer needs to check every time.

Reply:
I have just investigated and found that go does not have "contains" functionality built-in, and you would need to iterate over the entire array again to check.  The current approach is straightforward enough to keep.  You can resolve this thread with no changes.